src/mrsimulator/methods/base.py

```diff
@@ -24,95 +20,86 @@ class BaseMethod(Method):
     ndim: ClassVar[int] = 1
 
     def __init__(self, **kwargs):
-        BaseMethod.check(kwargs, self.__class__.ndim)
         super().__init__(**kwargs)
 
```

Comment:
There is need for the `BaseMethod` class. It does nothing besides passing the args to the parent `Method` class.
Remove this and replace `class BaseNamedMethod(BaseMethod)` with `class BaseNamedMethod(Method)`